Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -135,6 +135,58 @@ namespace AZ::Dom
             },
             m_value);
     }
+
+    bool operator<(const AZ::Dom::PathEntry& lhs, const AZ::Dom::PathEntry& rhs)
+    {
+        if (lhs.IsIndex())
+        {
+            return lhs.GetIndex() < rhs;
+        }
+        else
+        {
+            return lhs.GetKey() < rhs;
+        }
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, size_t index)
+    {
+        return entry.IsIndex() && entry.GetIndex() < index;
+    }
+
+    bool operator<(size_t index, const AZ::Dom::PathEntry& entry)
+    {
+        return !entry.IsIndex() || index < entry.GetIndex();
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, const AZ::Name& key)
+    {
+        return entry < key.GetStringView();
+    }
+
+    bool operator<(const AZ::Name& key, const AZ::Dom::PathEntry& entry)
+    {
+        return key.GetStringView() < entry;
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, AZStd::string_view key)
+    {
+        return entry.IsIndex() ||
+            AZStd::lexicographical_compare(
+                entry.GetKey().GetStringView().begin(),
+                entry.GetKey().GetStringView().end(),
+                key.begin(),
+                key.end());
+    }
+
+    bool operator<(AZStd::string_view key, const AZ::Dom::PathEntry& entry)
+    {
+        return entry.IsKey() &&
+            AZStd::lexicographical_compare(
+                key.begin(),
+                key.end(),
+                entry.GetKey().GetStringView().begin(),
+                entry.GetKey().GetStringView().end());
```

Comment:
repeat: This should use the string_view operator< free function instead of lexicographical_compare
```suggestion
                return entry.isKey() && key < entry.GetKey().GetStringView();
```

Reply:
Same as above. Swapping this out.